### PR TITLE
Do not broadcast queries if Apollo state has not changed and if the previousState actually has anything inside of it.

### PR DESCRIPTION
**Pull request by abhiaiyer91**
TODO:

- [x] Update CHANGELOG.md with your change
- [x] Make sure all of the significant new logic is covered by tests
- [x] Rebase your changes on master so that they can be merged easily
- [x] Make sure all tests and linter rules pass

### CODE REVIEW

**Comment by abhiaiyer91** [comment]
actually i need to deepequal

**Comment by stubailo** [comment]
Couldn't we just do `===` because of the immutable nature of the store? I feel like `deepEqual`-ing the entire store isn't necessary.

**Comment by abhiaiyer91** [comment]
You may be right! I was just thinking that the store state is essentially

```
{
   queries,
   mutations,
   data
}
```

if we do a straight up `===` will we know if things actually changed? I guess I should have tested that.

**Comment by stubailo** [comment]
Ohhhhh, wait you're right - I don't think we currently ensure that the old object is the same as the new one in our top-level reducer.

So for now let's do deepEqual and see what happens, we can optimize later. Maybe add a XXX comment or file a `perf` issue about it.

**Comment by abhiaiyer91** [comment]
Ill add a comment and update the change log later

**Comment by stubailo** [comment]
Btw we should probably add at least one test that checks that the observer doesn't fire if the result didn't change!

**Comment by abhiaiyer91** [comment]
Will do.

**Comment by abhiaiyer91** [comment]
Yeeeeere

File: test/QueryManager.ts [hunk 1]
```diff
@@ -1153,6 +1157,164 @@ describe('QueryManager', () => {
...
+          });
+        }
+        assert.equal(handleCount, 2);
+        done();
+      },
+    });
+
+  });
+
+  it('allows you to refetch queries with new variables', (done) => {
```
Comment (stubailo) [inline test/QueryManager.ts]: How did this test get into this PR? Is this a new test or did it somehow get duplicated accidentally?

**Comment by stubailo** [comment]
Looks awesome! Just confused about that extra test that seems unrelated.

File: test/QueryManager.ts (hunk 1 above)
Comment (abhiaiyer91) [inline test/QueryManager.ts]: WHOA DUPE

**Comment by stubailo** [comment]
LGTM :shipit:

**Comment by stubailo** [comment]
Great contribution dude!

**Pull request merged**